cpp/CMakeLists.txt

```diff
@@ -266,6 +266,16 @@ add_custom_command(OUTPUT INSTALL_PYTHON_CFFI
 
 add_custom_target(install_python DEPENDS cudf rmm_install_python PYTHON_CFFI INSTALL_PYTHON_CFFI)
 
+	###################################################################################################
+# - make documentation ----------------------------------------------------------------------------
+
+add_custom_command(OUTPUT CUDF_DOXYGEN
+                   WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/doxygen
+                   COMMAND doxygen Doxyfile
+                   VERBATIM)
+
+add_custom_target(doc DEPENDS CUDF_DOXYGEN)
```

Comment:
I think @jrhemstad added a `make doc` target in another outstanding PR. #823 ?

Reply:
This has his pr merged into it. Is that terrible? I wanted to make sure my changes work I can remove the dependency on his pr if you like.